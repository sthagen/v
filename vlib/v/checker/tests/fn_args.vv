fn u8(a byte) {}
fn arr(a []int) {}
fn fun(a fn(int)) {}

fn basic() {
	v := 4
	u8(&v)
	arr([5]!)
	fun(fn(i &int){})
}

struct S1 {
}
fn f(p &S1) {}

fn ptr() {
	v := 4
	f(v)
}
