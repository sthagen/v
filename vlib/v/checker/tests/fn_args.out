vlib/v/checker/tests/fn_args.vv:7:5: error: cannot use `&int` as `byte` in argument 1 to `u8`
    5 | fn basic() {
    6 |     v := 4
    7 |     u8(&v)
      |        ~~
    8 |     arr([5]!)
    9 |     fun(fn(i &int){})
vlib/v/checker/tests/fn_args.vv:8:6: error: cannot use `[1]int` as `[]int` in argument 1 to `arr`
    6 |     v := 4
    7 |     u8(&v)
    8 |     arr([5]!)
      |         ~~~~
    9 |     fun(fn(i &int){})
   10 | }
vlib/v/checker/tests/fn_args.vv:9:6: error: cannot use `fn (&int)` as `fn (int)` in argument 1 to `fun`
    7 |     u8(&v)
    8 |     arr([5]!)
    9 |     fun(fn(i &int){})
      |         ~~~~~~~~~~~~
   10 | }
   11 |
Details: ``'s expected fn argument: `` is NOT a pointer, but the passed fn argument: `i` is a pointer
vlib/v/checker/tests/fn_args.vv:18:4: error: cannot use `int` as `&S1` in argument 1 to `f`
   16 | fn ptr() {
   17 |     v := 4
   18 |     f(v)
      |       ^
   19 | }
